node/core/candidate-selection/src/lib.rs

```diff
@@ -206,6 +281,16 @@ impl CandidateSelectionJob {
 	) {
 		let _timer = self.metrics.time_handle_collation();
 
+		if self.assignment != para_id {
+			tracing::info!(
+				target: LOG_TARGET,
+				"Collator {:?} sent a collation outside of our assignment {:?}",
+				collator_id,
+				para_id,
+			);
+			return;
```

Comment:
probably we should report or disconnect from this collator

Reply:
Yeah. Collators shouldn't be advertising to validators who aren't in their group.